inject-java-test/src/test/groovy/io/micronaut/aop/MyInterface.java

```diff
@@ -0,0 +1,16 @@
+package io.micronaut.aop;
+
+interface MyInterface {
+
+    String process(String str);
+
+    // This method will not be added to the generated intercepted class for some reason.
+    String process(String str, int intParam);
+
+    // This one will
+    String process2(String str, int intParam);
+
```

Comment:
Ok, created repro without `Connection` as this reproduces with any interface. Seems like method `process(String, int)` clashes with `process(String, int[])` and the one with array gets created and one with just int gets skipped for some reason.

Reply:
Because I wrote it maybe the inverse way, wanting to prove that AbstractError is thrown (this means that interface does not implement expected method)
```
when:"Intercepted MyInterface does not have method implemented"
        myInterface.process("abc", 1)
        then:
        AbstractMethodError ex = thrown()
        ex
```
this should not be thrown if method is generated properly in the interceptor

Reply:
Basically the same happening with Connection `PreparedStatement prepareStatement(String sql, int autoGeneratedKeys)`